plugins/yaml/src/org/jetbrains/yaml/references/YAMLAliasReference.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2000-2017 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jetbrains.yaml.references;
+
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiReferenceBase;
+import com.intellij.psi.util.PsiTreeUtil;
+import com.intellij.util.ArrayUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.yaml.psi.YAMLScalar;
+
+import java.util.Collection;
+import java.util.Comparator;
+
+public class YAMLAliasReference extends PsiReferenceBase<PsiElement> {
+
+  private static final String ANCHOR_PREFIX = "&";
+
+  public YAMLAliasReference(@NotNull PsiElement element) {
+    super(element);
+  }
+
+  @Nullable
+  @Override
+  public PsiElement resolve() {
+    Collection<YAMLScalar> values = PsiTreeUtil.findChildrenOfType(myElement.getContainingFile(), YAMLScalar.class);
```

Comment:
Yes, I also had similar thoughts after seeing the YT ticket; this needs to be solved more comprehensively. I do like the idea of trying out lexer and parser, I never had a real need until now :) Let me check them out and write up a proposal how this could be done. This might take few weeks though as I have my daily work as well!